src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/LoopDetector.kt

```diff
@@ -19,7 +23,19 @@ import java.util.ArrayList
  *
  * The LoopDetector functions in two modes: default and replay mode.
  *
- * In default mode:
+ * In the default mode:
+ *
+ * In the default mode, LoopDetector has two states.
+ * When a spin cycle is not detected yet, LoopDetector ignores method parameters and read/write values and receivers,
+ * operating only with switch points and code locations.
+ * When a spin cycle is detected, LoopDetector requests the replay of the execution and starts tracking
+ * the parameters, etc., to detect a spin cycle period properly, taking into account all changes during the spin cycle.
+ * When we run into a spin cycle again, LoopDetector tries to find the period using [currentThreadCodeLocationsHistory].
+ * Parameters etc. are converted to the negative integer representation using hashcode (see [paramToIntRepresentation]).
```

Comment:
You now use the `CodeIdentity` interface, please update the documentation accordingly.